test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetTag.java

```diff
@@ -84,144 +84,237 @@ public static void main(String... args) throws Exception {
+//                    }
+//                    """, "foo6", null),
+
+                new SnippetAttributes("""
+                    {@snippet id="" :
+                        Hello, Snippet!
+                    }
+                    """, null, null),
+                new SnippetAttributes("""
+                    {@snippet id="":
+                        Hello, Snippet!
+                    }
+                    """, null, null),
+                new SnippetAttributes("""
+                    {@snippet id='':
+                        Hello, Snippet!
+                    }
+                    """, null, null),
+                new SnippetAttributes("""
+                    {@snippet id=:
+                        Hello, Snippet!
+                    }
+                    """, null, null),
+                new SnippetAttributes("""
+                    {@snippet lang="java" :
+                        Hello, Snippet!
+                    }
+                    """, null, "java"),
+                new SnippetAttributes("""
+                    {@snippet lang="java":
+                        Hello, Snippet!
+                    }
+                    """, null, "java"),
+                new SnippetAttributes("""
+                    {@snippet lang='java' :
+                        Hello, Snippet!
+                    }
+                    """, null, "java"),
+                new SnippetAttributes("""
+                    {@snippet lang='java':
+                        Hello, Snippet!
+                    }
+                    """, null, "java"),
+                new SnippetAttributes("""
+                    {@snippet lang=java :
+                        Hello, Snippet!
+                    }
+                    """, null, "java"),
+                new SnippetAttributes("""
+                    {@snippet lang="properties" :
+                        Hello, Snippet!
+                    }
+                    """, null, "properties"),
+                new SnippetAttributes("""
+                    {@snippet lang="text" :
+                        Hello, Snippet!
+                    }
+                    """, null, "text"),
+                new SnippetAttributes("""
+                    {@snippet lang="" :
+                        Hello, Snippet!
+                    }
+                    """, null, null),
+                new SnippetAttributes("""
+                    {@snippet lang="foo" id="bar" :
+                        Hello, Snippet!
+                    }
+                    """, "bar", "foo")
         );
         ClassBuilder classBuilder = new ClassBuilder(tb, "pkg.A")
                 .setModifiers("public", "class");
         forEachNumbered(snippets, (s, i) -> {
             classBuilder.addMembers(
                     MethodBuilder.parse("public void case%s() { }".formatted(i))
-                            .setComments(s));
+                            .setComments("A method.", s.content()));
```

Comment:
Did you add that "A method." to increase test specificity?

Reply:
I added the first "sentence" to have the snippet appear only in the details section, rather than both the summary and the detail sections. Since some of the snippets have id attributes, having the snippets rendered twice with a page would result in invalid HTML. I think it is acceptable behavior to not implement some "only render id once per page" logic, but it's good to bring it up and discuss it, so I'm glad you spotted this.